apps/src/netsim/NetSimSendWidget.js

```diff
@@ -48,30 +81,137 @@ NetSimSendWidget.createWithin = function (element, connection) {
 };
 
 /**
+ * Creates a keyPress handler that allows only the given characters to be
+ * typed into a text field.
+ * @param {RegExp} whitelistRegex
+ * @return {function} appropriate to pass to .keypress()
+ */
+var whitelistCharacters = function (whitelistRegex) {
```

Comment:
Of course, I break this same rule at the top of the file!  I'll at least change things to be consistent.